internal/contour/endpointstranslator.go

```diff
@@ -182,12 +167,12 @@ func (e *EndpointsTranslator) recomputeClusterLoadAssignment(oldep, newep *v1.En
 		}
 	}
 
+	e.SnapshotHandler.UpdateSnapshot()
```

Comment:
Does this change the frequency at which we update Envoy? IIUC every endpoint change will do a full snapshot; is that what we want here?

Reply:
That's what happens today, any endpoint change in Kubernetes updates EDS with that new endpoint. I have another issue that looks to reduce this churn since every endpoint ends up in EDS today which is overkill for the amount of endpoints that actually need to be in EDS. 

The alternative is to put updates to EDS in the same loop as the dag rebuild, but I don't think that's a good idea as you'll be delaying updates to EDS which is why it's been kept separate, we want changes to update asap in that cache.

Reply:
> That's what happens today, any endpoint change in Kubernetes updates EDS with that new endpoint. I have another issue that looks to reduce this churn since every endpoint ends up in EDS today which is overkill for the amount of endpoints that actually need to be in EDS.

Yeh, but with this change, a new endpoint increments the version of every other resource too, which IIUC means they would get sent to Envoy.

Reply:
I can give endpoints their own version separate from the other services. Is that what you are asking for? 

The way it is now if Envoy receives the same version, it ignores the change.

Reply:
I'm asking about the implications and effects of this change. IIUC, we handle endpoints specially due to the rate of change. If we stop doing what, what happens?

Reply:
Are you suggesting that we handle endpoints differently now with go-control-plane? Previously we updated per endpoint and this is what the PR does now. 

The alternative is to tie endpoint updates into the dag rebuild time (or some other backoff) which would delay updates to Envoy and not what we want since those endpoints can change and if we don't update Envoy, users will get errors.